vespalib/src/vespa/vespalib/util/memory.h

```diff
@@ -11,6 +11,70 @@
 
 namespace vespalib {
 
+inline void *memcpy_safe(void *dest, const void *src, size_t n) noexcept {
+    if (n > 0) [[likely]] {
```

Comment:
I prefer having the non likely if be guards instead.
In this case 
if (n==0) { return dest; }
 return memcpy(...);
The it is easier to follow the happy fast path which will be without indentation.